drivers/include/sht2x.h

```diff
@@ -0,0 +1,194 @@
+#define SHT2X_USER_RESOLUTION_MASK          0x81
+#define SHT2X_USER_EOB_MASK                 0x40
+#define SHT2X_USER_HEATER_MASK              0x04
+#define SHT2X_USER_RESERVED_MASK            0x38
+#define SHT2X_USER_OTP_MASK                 0x02
+/** @} */
+
+/**
+ * @brief       Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                      /**< I2C device               */
+    uint8_t i2c_addr;                   /**< I2C address              */
+    sht2x_res_t resolution;             /**< resolution bits RH/temp  */
+    sht2x_measure_mode_t measure_mode;  /**< measurement mode used    */
+    bool is_crc_enabled;                /**< do CRC or not            */
+} sht2x_params_t;
+
+/**
+ * @brief       Device descriptor for the SHT2X sensor
+ */
+typedef struct {
+    sht2x_params_t params;              /**< Device Parameters */
+} sht2x_t;
+
+/**
+ * @brief       Initialize the given SHT2X device
+ *
+ * @param[out]  dev         Initialized device descriptor of SHT2X device
+ * @param[in]   params      The parameters for the SHT2x device
+ *
+ * @return                  SHT2X_OK on success
+ * @return                  SHT2X_ERR_I2C if given I2C is not enabled in board config
+ */
+int sht2x_init(sht2x_t* dev, const sht2x_params_t* params);
+
+/**
+ * @brief       Reset the SHT2X device
+ *
+ * @param[in]   dev         Device descriptor of SHT2X device to read from
+ *
+ * @return                  SHT2X_OK on success
+ * @return                  SHT2X_ERR_I2C if given I2C is not enabled in board config
+ */
+int sht2x_reset(sht2x_t* dev);
+
+/**
+ * @brief       Read temperature value from the given SHT2X device, returned in
+ *              centi °C
+ *
+ * @param[in]  dev          Device descriptor of SHT2X device to read from
+ *
+ * @returns                 The temperature in centi Celsius. In case of an error
+ *                          it returns INT16_MIN.
+ */
+int16_t sht2x_read_temperature(const sht2x_t* dev);
+
+/**
+ * @brief       Read humidity value from the given SHT2X device, returned in
+ *              centi %RH
+ *
+ * @details     This function should only be called after doing bmx280_read_temperature
+ *              first. It's only available with SHT2X sensor.
+ *
+ * @param[in]  dev          Device descriptor of SHT2X device to read from
+ *
+ * @returns                 Humidity in centi %RH (i.e. the percentage times 100).
+ *                          In case of an error it returns 0 (i.e. an unrealistic value).
+ */
+uint16_t sht2x_read_humidity(const sht2x_t *dev);
+
+/**
+ * @brief       Read identification code from the given SHT2X device
+ *
+ * @param[in]  dev          Device descriptor of SHT2X device to read from
+ * @param[out] buffer       Buffer to write the identification code to
+ * @param[in]  buflen       The size of @p buffer
+ *
+ * @return                  >= 0 on success, number of bytes read
+ * @return                  -1 if the code could not be acquired
```

Comment:
Why `-1` not value from the enum above?

Reply:
OK. I'll add some more error codes and use these